src/OpenLoco/Paint/PaintEntity.cpp

```diff
@@ -32,7 +33,153 @@ namespace OpenLoco::Paint
         regs.dx = base->z;
         regs.ebx = (base->sprite_yaw + (session.getRotation() << 4)) & 0x3F;
         regs.esi = reinterpret_cast<int32_t>(base);
-        call(0x00440325, regs);
+        //call(0x00440325, regs);
+        //return;
+
+        Gfx::drawpixelinfo_t* dpi = session.getContext();
+
+        switch (base->getSubType()) // 00440326-0044032A
+        {
+            case MiscThingType::exhaust: // 0
+            {
+                if (dpi->zoom_level > 1) // 00440331-0044033C
+                {
+                    return;
+                }
+                Exhaust* exhaustObject = base->as_exhaust();
+                steam_object* steamObject = ObjectManager::get<steam_object>(exhaustObject->object_id & 0x7f); // 00440342-00440349
+                static_assert(offsetof(steam_object, var_16) == 0x16);
+                static_assert(offsetof(steam_object, var_1A) == 0x1A);
+                static_assert(offsetof(steam_object, sound_effect) == 0x1E);
+
+                uint8_t* edi = reinterpret_cast<uint8_t*>((exhaustObject->object_id & 0x80) == 0 ? steamObject->var_16 : steamObject->var_1A); // 00440354-0044035D
+                uint32_t imageId = edi[2 * exhaustObject->var_26];                                                                             // 00440350, 00440360
+                imageId = imageId + steamObject->var_0E + steamObject->var_0A;                                                                 // 00440364-00440367
+
+                if ((steamObject->var_08 & 8) != 0) // 0044036A-00440370
+                {
+                    session.addToPlotList2(imageId, { 0, 0, base->z }, { 1, 1, 0 }); // 00440372-00440386
+                }
+                else
+                {
+                    session.addToPlotListAsParent(imageId, { 0, 0, base->z }, { 1, 1, 0 }, { -12, -12, base->z }); // 0044038F-004403BC
+                }
+                break;
+            }
+
+            case MiscThingType::redGreenCurrency: // 1
+            {
+                if (dpi->zoom_level > 1) // 004403C5-004403D0
+                {
+                    return;
+                }
+                MoneyEffect* moneyEffect = base->asRedGreenCurrency();
+                const string_id stringId = moneyEffect->amount >= 0 ? StringIds::format_currency_income_green : StringIds::format_currency_expense_red_negative; // 004403D2-004403DF
+                uint32_t currencyAmount = abs(moneyEffect->amount);
+                uint32_t yOffsets = 0x4FAAC8 + moneyEffect->wiggle; // 004403E3
```

Comment:
give 0x4FAAC8 a name like `_wiggleYOffsets` or something.

Reply:
Named as `wiggleYOffsets` (before the switch statement)